### PR TITLE
Fix maxPageSize parameter

### PR DESCRIPTION
**Pull request by welovej**
# Description
<i>Fix issue : https://github.com/Azure/azure-sdk-for-net/issues/44735   which maxPageSieParameter was lost</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by welovej** [comment]
To fix the Issue : [The pagesize parameter is not taken effect during the pageable operation #44735](https://github.com/Azure/azure-sdk-for-net/issues/44735)

File: src/assets/Generator.Shared/GeneratorPageableHelpers.cs [hunk 1]
```diff
@@ -41,7 +41,20 @@ public static AsyncPageable<T> CreateAsyncPageable<T>(
...
+        public static AsyncPageable<T> CreateAsyncPageable<T>(
+            Func<int?, HttpMessage>? createFirstPageRequest,
+            Func<int?, string, HttpMessage>? createNextPageRequest,
+            Func<JsonElement, T> valueFactory,
+            ClientDiagnostics clientDiagnostics,
+            HttpPipeline pipeline,
+            string scopeName,
+            string? itemPropertyName,
+            string? nextLinkPropertyName,
+            object defaultPageSize,
```
Comment (live1206) [inline src/assets/Generator.Shared/GeneratorPageableHelpers.cs:53]: This should be `int`, we don't allow run-time casting during generation.

Comment (welovej) [reply src/assets/Generator.Shared/GeneratorPageableHelpers.cs]: Fixed

File: src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs [hunk 2]
```diff
@@ -147,12 +147,17 @@ public static CodeWriter WritePageableBody(this CodeWriter writer, IReadOnlyList
...
-        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName)
+        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName, bool isCreateListReques = false)
         {
             createPageableParameters.Add($"{scopeName:L}");
             createPageableParameters.Add($"{itemPropertyName:L}");
             createPageableParameters.Add($"{nextLinkPropertyName:L}");
-
+            string? maxPageSize = methodParameters.Where(s => s.Name == "maxpagesize").FirstOrDefault()?.Name;
+            string? maxPageSizeType = methodParameters.Where(s => s.Name == "maxpagesize").FirstOrDefault()?.Type.Name;
+            if (maxPageSize != null && maxPageSizeType != "String" && isCreateListReques)
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs:157]: We should check if maxPageSizeType is `int` directly.

File: src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs [hunk 3]
```diff
@@ -131,8 +131,8 @@ public static CodeWriter WritePageableBody(this CodeWriter writer, IReadOnlyList
                 clientDiagnosticsReference.GetReferenceFormattable(),
                 pipelineReference.GetReferenceFormattable()
             };
-
-            createPageableParameters.AddTrailingPageableParameters(methodParameters, scopeName, itemPropertyName, nextLinkPropertyName);
+            bool isCreateListRequest = firstPageRequest?.GetArguments()[1]?.ToString() == "List";
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs:134]: why is this needed?

File: src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs [hunk 4]
```diff
@@ -147,12 +147,17 @@ public static CodeWriter WritePageableBody(this CodeWriter writer, IReadOnlyList
...
         }
 
-        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName)
+        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName, bool isCreateListReques = false)
         {
             createPageableParameters.Add($"{scopeName:L}");
             createPageableParameters.Add($"{itemPropertyName:L}");
             createPageableParameters.Add($"{nextLinkPropertyName:L}");
-
+            string? maxPageSize = methodParameters.Where(s => s.Name == "maxpagesize").FirstOrDefault()?.Name;
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs:155]: put `mapagesize` to [KnowParameters ](https://github.com/Azure/autorest.csharp/blob/main/src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs)

File: src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs [hunk 5]
```diff
@@ -58,5 +60,9 @@ internal static partial class KnownParameters
             DefaultValue: null,
             Validation: ValidationType.None,
             Initializer: null);
+        public static bool TypeAndNameMatch(Parameter parameter1, Parameter parameter2)
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs:63]: is this needed? can we use the existing one？
https://github.com/Azure/autorest.csharp/blob/268545602af22109c028c7c7316fc15a056c1d12/src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs#L25

**Review by live1206 — APPROVED** [review]
💯
